### PR TITLE
Don't import inputs not sent to the current wallet

### LINKED ISSUE #54: The testing client imports unspendable inputs
### Affected Branch

trunk

### Basic Diagnostics

- [X] I've pulled the latest changes on the affected branch and the issue is still present.

- [X] The issue is reproducible in docker

### Description

root@2af70284b1b5:/opt/tx-processor#  ./build/src/uhs/client/client-cli 2pc-compose.cfg mempool0.dat wallet0.dat mint **1 1**
[2022-02-13 03:59:18.147] [WARN ] Existing wallet file not found
[2022-02-13 03:59:18.147] [WARN ] Existing client file not found
fdcc8e5559af55040a4f0f12db2b4de5f614ffaf08d359d727b3b07c12a6c64f
root@2af70284b1b5:/opt/tx-processor#  ./build/src/uhs/client/client-cli 2pc-compose.cfg mempool1.dat wallet1.dat newaddress
[2022-02-13 03:59:54.151] [WARN ] Existing wallet file not found
[2022-02-13 03:59:54.152] [WARN ] Existing client file not found
usd1qpsctc3vzrwa2e2ru0lehgjh8p5l8y2cklsuh6sd6grr3n6eqayn7r9zgq8
root@2af70284b1b5:/opt/tx-processor#  ./build/src/uhs/client/client-cli 2pc-compose.cfg mempool0.dat wallet0.dat send 1 usd1qpsctc3vzrwa2e2ru0lehgjh8p5l8y2cklsuh6sd6grr3n6eqayn7r9zgq8
tx_id:
f75737d445cdf02ba8ca7cf6372930cc974c7e9cbc2ef3fc814b379772cdecf5
Data for recipient importinput:
f75737d445cdf02ba8ca7cf6372930cc974c7e9cbc2ef3fc814b379772cdecf50000000000000000e5ad0c7243ec63ff655e959f7ed68be9eebec3c1b38e1c11956a37207956f5f00100000000000000
Sentinel responded: Confirmed
root@2af70284b1b5:/opt/tx-processor# ./build/src/uhs/client/client-cli 2pc-compose.cfg mempool1.dat wallet1.dat importinput f75737d445cdf02ba8ca7cf6372930cc974c7e9cbc2ef3fc814b379772cdecf50000000000000000e5ad0c7243ec63ff655e959f7ed68be9eebec3c1b38e1c11956a37207956f5f00100000000000000
root@2af70284b1b5:/opt/tx-processor# ./build/src/uhs/client/client-cli 2pc-compose.cfg mempool0.dat wallet0.dat importinput f75737d445cdf02ba8ca7cf6372930cc974c7e9cbc2ef3fc814b379772cdecf50000000000000000e5ad0c7243ec63ff655e959f7ed68be9eebec3c1b38e1c11956a37207956f5f00100000000000000
root@2af70284b1b5:/opt/tx-processor#  ./build/src/uhs/client/client-cli 2pc-compose.cfg mempool1.dat wallet1.dat sync
root@2af70284b1b5:/opt/tx-processor# ./build/src/uhs/client/client-cli 2pc-compose.cfg mempool0.dat wallet0.dat sync
root@2af70284b1b5:/opt/tx-processor#  ./build/src/uhs/client/client-cli 2pc-compose.cfg mempool1.dat wallet1.dat info
**Balance: $0.01**, UTXOs: 1, pending TXs: 0
root@2af70284b1b5:/opt/tx-processor# ./build/src/uhs/client/client-cli 2pc-compose.cfg mempool0.dat wallet0.dat info
**Balance: $0.01**, UTXOs: 1, pending TXs: 0

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by jonwiggins**
Hi,

I took at swing at fixing #54. I did this by mirroring how a similar check is done in `transaction::wallet::sign`. I thought about splitting it off into a common method that could be both places, but I didn't want to mess with the other operations done with the lock in the `sign` method.

The new output from the steps shown in the issue is:

```
root@9c5f6600cab3:/opt/tx-processor# ./build/src/uhs/client/client-cli 2pc-compose.cfg mempool0.dat wallet0.dat mint 1 1
[2022-03-15 04:41:34.807] [WARN ] Existing wallet file not found
[2022-03-15 04:41:34.807] [WARN ] Existing client file not found
040e6a5e3e0b7ed30a61696693ace87abb7444e0fe9bb0fa34208f52a0dcb40a
root@9c5f6600cab3:/opt/tx-processor# ./build/src/uhs/client/client-cli 2pc-compose.cfg mempool1.dat wallet1.dat newaddress
[2022-03-15 04:41:39.851] [WARN ] Existing wallet file not found
[2022-03-15 04:41:39.851] [WARN ] Existing client file not found
usd1qrefhljq2n73vyhc0vf85cack6685cze704qeqs32kjdhaku68yhcwnceqc
root@9c5f6600cab3:/opt/tx-processor# ./build/src/uhs/client/client-cli 2pc-compose.cfg mempool0.dat wallet0.dat send 1 usd1qrefhljq2n73vyhc0vf85cack6685cze704qeqs32kjdhaku68yhcwnceqc
tx_id:
a93ae49bc9e2bd50f1f231ec57a7928111568fb89bef1d38096ca6d2a835bc4c
Data for recipient importinput:
a93ae49bc9e2bd50f1f231ec57a7928111568fb89bef1d38096ca6d2a835bc4c00000000000000005a387c6a6eb9748158550430f1536efc993932936c603e72cc6dd26fb43a881c0100000000000000
Sentinel responded: Confirmed
root@9c5f6600cab3:/opt/tx-processor# ./build/src/uhs/client/client-cli 2pc-compose.cfg mempool1.dat wallet1.dat importinput a93ae49bc9e2bd50f1f231ec57a7928111568fb89bef1d38096ca6d2a835bc4c00000000000000005a387c6a6eb9748158550430f1536efc993932936c603e72cc6dd26fb43a881c0100000000000000
root@9c5f6600cab3:/opt/tx-processor# ./build/src/uhs/client/client-cli 2pc-compose.cfg mempool0.dat wallet0.dat importinput a93ae49bc9e2bd50f1f231ec57a7928111568fb89bef1d38096ca6d2a835bc4c00000000000000005a387c6a6eb9748158550430f1536efc993932936c603e72cc6dd26fb43a881c0100000000000000
[2022-03-15 04:42:27.411] [WARN ] Ignoring non-spendable input
root@9c5f6600cab3:/opt/tx-processor# ./build/src/uhs/client/client-cli 2pc-compose.cfg mempool1.dat wallet1.dat sync
root@9c5f6600cab3:/opt/tx-processor# ./build/src/uhs/client/client-cli 2pc-compose.cfg mempool0.dat wallet0.dat sync
root@9c5f6600cab3:/opt/tx-processor# ./build/src/uhs/client/client-cli 2pc-compose.cfg mempool1.dat wallet1.dat info
Balance: $0.01, UTXOs: 1, pending TXs: 0
root@9c5f6600cab3:/opt/tx-processor# ./build/src/uhs/client/client-cli 2pc-compose.cfg mempool0.dat wallet0.dat info
Balance: $0.00, UTXOs: 0, pending TXs: 0
```

Let me know what you think. I'm happy to make any needed changes if this seems like a plausible solution.

Thanks

Signed-off-by: Jon Wiggins <me@jonwiggins.org>

### CODE REVIEW

**Comment by HalosGhost** [comment]
Welcome! I will dive in and review today.

**Review by HalosGhost — CHANGES_REQUESTED** [review]
Pulling and testing locally, this looks like it solves the issue quite well. So, I only have a couple of small logistical requests:

* Can you rephrase and shorten your commit header (less than 50 characters)? I might suggest something like “fix: Avoid importing unspendable inputs” though the new title for the PR you chose would also be fine
* Can you include `closes #54` in the body of the commit message? (this will result in the issue auto-closing when this gets merged.)

More information on why we prefer these kinds of conventions can be found in our [getting started guide](https://github.com/mit-dci/opencbdc-tx/blob/trunk/docs/getting-started.md#what-does-a-good-commit-look-like).

Aside from these small issues, this looks like an excellent first contribution, and I look forward to merging it!

**Comment by jonwiggins** [comment]
Great, thanks.
I've made those changes. And I'll keep an eye out for future issues I think I can help to fix.

**Comment by HalosGhost** [comment]
> I've made those changes.

Thank you! Sorry to ask for another small change, but each section of the commit message needs to be separated by an empty line (namely, the `Signed-off-by` trailer needs to have a newline between it and the message body; similarly, the commit message header needs a newline between it and the message body).

**Comment by jonwiggins** [comment]
Whoops, sorry about that.

How about now?

**Comment by HalosGhost** [comment]
Perfect. I'll let the CI run one last time and then merge it today!

**Review by HalosGhost — APPROVED** [review]
Very simple and clean check to only import an input into a wallet if the input is spendable by the wallet.

Looks good to me!
